source/h2_stream.c

```diff
@@ -535,20 +536,46 @@ int aws_h2_stream_on_decoder_push_promise(struct aws_h2_stream *stream, uint32_t
         s_get_h2_connection(stream), promised_stream_id, AWS_H2_ERR_REFUSED_STREAM);
 }
 
-int aws_h2_stream_on_decoder_data(struct aws_h2_stream *stream, struct aws_byte_cursor data) {
+int aws_h2_stream_on_decoder_data_begin(struct aws_h2_stream *stream, uint32_t data_payload_len) {
     AWS_PRECONDITION_ON_CHANNEL_THREAD(stream);
 
+    if (stream->thread_data.window_size_self < 0 ||
+        (stream->thread_data.window_size_self == 0 && data_payload_len > 0)) {
```

Comment:
It's probably OK to get a DATA frame with 0 payload_len even if the stream's window is < 0.
right?

Reply:
nevermind, we talked and the other way is too confusing to be worth it

Reply:
I don't really know... But spec says "A sender MUST track the negative flow-control window and MUST NOT send new flow-controlled frames until it receives WINDOW_UPDATE frames that cause the flow-control window to become positive." MUST NOT send new flow-controlled frames